### PR TITLE
feat: add delivery bundles usecheckout

### PR DESCRIPTION
**Pull request by escabora**
## Description

We understand that the responsibility for exposing delivery bundles data lies with `useCheckout`. So I opened this PR in which I get the delivery bundles in the context of redux and expose it in the `useCheckout` hook. I also take the opportunity to add a key to the `useCheckout` that represents which delivery bundle has been selected.

## Checklist

- [x] The commit message follows our guidelines
- [x] Tests for the respective changes have been added
- [x] The code is commented, particularly in hard-to-understand areas
- [x] The labels and/or milestones were added

## Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

**Review by boliveira — APPROVED** [review]

**Review by dnamorim — APPROVED** [review]

File: tests/__fixtures__/checkout/checkout.fixtures.mts [hunk 1]
```diff
@@ -1054,6 +1054,8 @@ export const deliveryBundlesEntity = {
   },
 };
 
+export const mockDeliveryBundlesEntityDenormalized = [deliveryBundle];
```
Comment (anapim) [inline tests/__fixtures__/checkout/checkout.fixtures.mts:1057]: This still doesn't look okay. Aren't we expecting two delivery bundles in the response?

**Review by anapim — CHANGES_REQUESTED** [review]

File: tests/__fixtures__/checkout/checkout.fixtures.mts (hunk 1 above)
Comment (ivomota) [reply tests/__fixtures__/checkout/checkout.fixtures.mts:1057]: I agree with @anapim. This seems to be not ok. 
Did you tried to add
```js
export const mockDeliveryBundlesEntityDenormalized = [
   deliveryBundle,
   {
    ...deliveryBundle,
    id: '090998',
    name: 'fake bundle',
    isSelected: false,
  }
];
```

I think this should be the result....

Comment (anapim) [reply tests/__fixtures__/checkout/checkout.fixtures.mts:1057]: I found the issue. On line 1000, we're missing the second delivery bundle id.

```javascript
export const checkoutEntity = {
  checkoutOrder: checkoutOrderId,
  id: checkoutId,
  shippingOptions: [shippingOption],
  deliveryBundles: [deliveryBundleId, '090998'],
  orderStatus: OrderStatusError.NoError,
};
```

Then @ivomota suggestions should work.

Comment (escabora) [reply tests/__fixtures__/checkout/checkout.fixtures.mts]: I made the adjustment. I also found the error in checkoutEntity (line 1000). I needed to add another delivery package id in checkoutEntity. However, when I run the test, I see an error in the redux checkout selectors. I have also solved this problem.

File: tests/__fixtures__/checkout/checkout.fixtures.mts [hunk 2]
```diff
@@ -997,7 +997,7 @@ export const checkoutEntity = {
   checkoutOrder: checkoutOrderId,
   id: checkoutId,
   shippingOptions: [shippingOption],
-  deliveryBundles: [deliveryBundleId],
+  deliveryBundles: [deliveryBundleId, '090998'],
```
Comment (dnamorim) [inline tests/__fixtures__/checkout/checkout.fixtures.mts:1000]: We are repeating this ID for a second `deliveryBundle` on our mock data several times and in the unit tests 

I would leave the suggestion to have two id's, as we have already one for the `deliveryBundleUpgradeId`:

```
export const deliveryBundleId_1 = '1234';
export const deliveryBundleId_2 = '090998';
```

Comment (escabora) [reply tests/__fixtures__/checkout/checkout.fixtures.mts:1000]: Adjustment made!

**Pull request merged**